src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -2833,38 +2717,35 @@ def create_run(self, pipeline_run: PipelineRunModel) -> PipelineRunModel:
                     )
 
             # Query stack
-            if pipeline_run.stack_id is not None:
-                stack = session.exec(
-                    select(StackSchema).where(
-                        StackSchema.id == pipeline_run.stack_id
-                    )
-                ).first()
-                if stack is None:
-                    logger.warning(
-                        f"No stack with ID '{pipeline_run.stack_id}' found. "
-                        f"Creating pipeline run '{pipeline_run.name}' without "
-                        "linked stack."
-                    )
-                    pipeline_run.stack_id = None
-
-            # Query pipeline
-            pipeline = None
-            if pipeline_run.pipeline_id is not None:
-                pipeline = session.exec(
-                    select(PipelineSchema).where(
-                        PipelineSchema.id == pipeline_run.pipeline_id
-                    )
-                ).first()
-                if pipeline is None:
-                    logger.warning(
-                        f"No pipeline with ID '{pipeline_run.pipeline_id}' "
-                        f"found. Creating pipeline run '{pipeline_run.name}' "
-                        f"as unlisted run."
-                    )
-                    pipeline_run.pipeline_id = None
+            if pipeline_run.stack is None:
+                logger.warning(
+                    f"No stack found for this run. "
+                    f"Creating pipeline run '{pipeline_run.name}' without "
+                    "linked stack."
+                )
 
-            new_run = PipelineRunSchema.from_create_model(
-                run=pipeline_run, pipeline=pipeline
+            if pipeline_run.pipeline is None:
+                logger.warning(
+                    f"No pipeline found. Creating pipeline run "
+                    f"'{pipeline_run.name}' as unlisted run."
+                )
```

Comment:
This is not what the old code was doing. It is fine if a user creates a run without a pipeline and we don't want to throw a warning every time then. What is problematic is if the user specifies a pipeline that doesn't exist anymore in which case this warning was thrown. This mainly happens if you run a pipeline and then you delete the pipeline before the run got synced from TFX into our DB.

Reply:
Fixed